scripts/cluster.lic

```diff
@@ -0,0 +1,179 @@
+require "dcell"
+
+module Cluster
+  ##
+  ## minimal options parser
+  ##
+  module Opts
+    FLAG_PREFIX    = "--"
+    
+    def self.parse_command(h, c)
+      h[c.to_sym] = true
+    end
+
+    def self.parse_flag(h, f)
+      (name, val) = f[2..-1].split("=")
+      if val.nil?
+        h[name.to_sym] = true
+      else
+        val = val.split(",")
+
+        h[name.to_sym] = val.size == 1 ? val.first : val
+      end
+    end
+
+    def self.parse(args = Script.current.vars[1..-1])    
+      OpenStruct.new(**args.to_a.reduce(Hash.new) do |opts, v|
+        if v.start_with?(FLAG_PREFIX)
+          Opts.parse_flag(opts, v)
+        else
+          Opts.parse_command(opts, v)
+        end
+        opts
+      end)
+    end
+
+    def self.method_missing(method, *args)
+      parse.send(method, *args)
+    end
+  end
+end
+##
+## distributed cluster for queryiny character states
+##
+module Cluster
+  ##
+  ## normalize a key in the cluster
+  ##
+  def self.key_of(t)
+    t.to_s.downcase.gsub(%{\s}, "_")
+  end
+  ##
+  ## all nodes
+  ##
+  def self.nodes()
+    DCell::Node.all
+  end
+  ##
+  ## fetch a Node by normalized key
+  ##
+  def self.node(id)
+    DCell::Node[key_of(id)]
+  end
+  ##
+  ## put a key to the dcell 
+  ## global k/v registry
+  ##
+  def self.put(key, value)
+    DCell::Global.set(key, value)
+  end
+  ##
+  ## fetch a value from the 
+  ## dcell global k/v registry
+  ##
+  def self.fetch(key, default = nil)
+    val = DCell::Global.get(key)
+    if val.nil?
+      default
+    else
+      val
+    end
+  end
+  ##
+  ## get all keys from the 
+  ## global k\v registry
+  ##
+  def self.keys()
+    DCell::Global.keys()
+  end
+  ##
+  ## check if a cluster implements
+  ## an actor everywhere
+  ##
+  def self.implements?(actor)
+    DCell::Node.all.map do |node|
+      [node.id, 
+        !node[actor].nil?]
+    end
+  end
+  ##
+  ## invoke a given method on a Node within an available Actor
+  ##
+  def self.call(id, method, *args, actor: :bridge)
+    if node = Cluster.get(id)
```

Comment:
I was getting: `Exception: private method `get' called for Cluster:Module` when using this, but it seems like it is working after I updated it to: `if node = Cluster.node(id)` instead.

Reply:
oh, good call, I changed it from `get` to `node` to be more explicit about what we are actually getting from the cluster